plugins/repository-s3/src/main/java/org/elasticsearch/cloud/aws/InternalAwsS3Service.java

```diff
@@ -126,17 +133,30 @@ public static ClientConfiguration buildConfiguration(Logger logger, Settings set
         return clientConfiguration;
     }
 
-    public static AWSCredentialsProvider buildCredentials(Logger logger, Settings settings, Settings repositorySettings) {
+    public static AWSCredentialsProvider buildCredentials(Logger logger, DeprecationLogger deprecationLogger,
+                                                          Settings settings, Settings repositorySettings) {
         AWSCredentialsProvider credentials;
         try (SecureString key = getValue(repositorySettings, settings,
                                          S3Repository.Repository.KEY_SETTING, S3Repository.Repositories.KEY_SETTING);
              SecureString secret = getValue(repositorySettings, settings,
                                             S3Repository.Repository.SECRET_SETTING, S3Repository.Repositories.SECRET_SETTING)) {
 
             if (key.length() == 0 && secret.length() == 0) {
-                // TODO: add deprecation, except for using instance profile
-                logger.debug("Using either environment variables, system properties or instance profile credentials");
-                credentials = new DefaultAWSCredentialsProviderChain();
+                // create a "manual" chain of providers here, so we can log deprecation of unsupported methods
+                AWSCredentials envCredentials = getDeprecatedCredentials(logger, deprecationLogger,
+                    new EnvironmentVariableCredentialsProvider(), "environment variables");
+                if (envCredentials != null) {
+                    credentials = new StaticCredentialsProvider(envCredentials);
+                } else {
+                    AWSCredentials syspropCredentials = getDeprecatedCredentials(logger, deprecationLogger,
+                        new SystemPropertiesCredentialsProvider(), "system properties");
+                    if (syspropCredentials != null) {
+                        credentials = new StaticCredentialsProvider(syspropCredentials);
+                    } else {
+                        logger.debug("Using instance profile credentials");
+                        credentials = new InstanceProfileCredentialsProvider();
```

Comment:
But if we run your code without setting any key/secret in `elasticsearch.yml` (or in the vault), then we will print the deprecated notice you wrote in `getDeprecatedCredentials`, right?

I mean that when you are using instance profile, you don't have to add any key/secret as it is automatically provided by AWS platform.